ext/xsl/php_xsl.stub.php

```diff
@@ -71,6 +71,12 @@
 
 class XSLTProcessor
 {
+    /** @var bool */
+    public $doXInclude;
```

Comment:
BTW: wouldn't it be better to add a dedicated property handler instead of having a property and a field in the internal structure?

Reply:
Thanks for checking.

> I'd strongly prefer to have native types :) For PHP 8.1, I declared type for basically all untyped properties where it was possible so it would be very unfortunate to break the trend. :)

Alright, just wasn't too sure regarding BC. I'll swap over to actual types and add a BC note to UPGRADING.
Also changed the code to use `zend_is_true` and use the R fetch instead of IS (as the property is now always defined).

> As far as I see, the correct type is?bool

I believe just a bool would do. Because:
1) For cloneDocument, the behaviour for false and null would be the same, so it makes sense to default the property to false.
2) For doXInclude, the default behaviour set out by libxslt is false, so it makes sense we default the property to false. Again like the previous item: null and false would be the same so null has no reason to exist :)

> but I'm not sure what we do with the xinclude field afterwards? Somehow I couldn't find any reference to it... 🤔

It's just passed to libxslt.

> BTW: wouldn't it be better to add a dedicated property handler instead of having a property and a field in the internal structure?

The internal structure you're referring to only exists temporarily during the xslt evaluation. While possible to add a handler, we'd have to store and manage the value ourselves into a non-temporary object. So I think that would only complicate things.

Reply:
> Alright, just wasn't too sure regarding BC. I'll swap over to actual types and add a BC note to UPGRADING.

IMO this is a small BC break which can easily be detected and fixed by end-users, so it should be fine to have even in a minor version. :)

And thanks for the rest of the answers!